app/webpacker/components/RegistrationsV2/Register/RegistrationOverview.jsx

```diff
@@ -37,15 +86,16 @@ export default function RegistrationOverview({
           {i18n.t('registrations.wont_pay_here')}
         </Message>
       )}
-      <Segment>
+      <Segment loading={isDeleting}>
         <Header>{i18n.t('competitions.nav.menu.registration')}</Header>
         <Form onSubmit={nextStep} size="large">
           <FormField>
             <label>
               {i18n.t('activerecord.attributes.registration.registration_competition_events')}
               :
             </label>
-            {registration.competing.event_ids.map((id) => (<EventIcon key={id} id={id} style={{ cursor: 'unset' }} />))}
+            { /* Make sure to keep WCA Event order */}
+            {events.official.flatMap((e) => (registration.competing.event_ids.includes(e.id) ? <EventIcon key={e.id} id={e.id} style={{ cursor: 'unset' }} /> : []))}
```

Comment:
Maybe do a `filter` first instead of this "hidden filter" by flat-mapping to an empty array?

Reply:
I chose to flat map so we only iterate over the array once, sadly JS does not have filterMap

Reply:
But you're not iterating over an excessively long array or over something that is costly to compute.

If every array iteration triggered lots and lots of DB calls, then fine. But here, we have a pre-computed list in-memory that is never longer than ~20 entries. Even doing one `includes` call per event is still max 400 calls which literally every processor of any computer less than 20 years old sneezes out in a time faster than I care to find the micro-femto-atto-prefix for :P

TL;DR: Readability and clarity goes over iteration optimization in simple cases like this

Reply:
changed!